Remove Seam Text

seanf opened the pull request:
https://bugzilla.redhat.com/show_bug.cgi?id=1198433

- [x] Implement CommonMark rendering for home/project about pages
- [x] Implement CommonMark preview for edit home/project content
- [x] Test conversion from Seam Text to CommonMark by comparing rendered HTML
- [x] Implement Liquibase database migration from Seam Text to CommonMark
- [x] Test Liquibase database migration

carlosmunoz commented on zanata-war/src/main/java/org/zanata/action/HomePage.java:
```diff
@@ -0,0 +1,67 @@
+    @In
+    private ApplicationConfiguration applicationConfiguration;
+
+    @In("commonMarkRenderer")
```
Any reason for using explicit names?

seanf replied:
I wanted the variable name to be shorter, without shortening the component
name.  The names aren't checked at compile time anyway, so I don't think we
lose anything.

The explicit name reference will go away with CDI anyway.

On 17 March 2015 at 11:53, Carlos Munoz <notifications@github.com> wrote:

> In zanata-war/src/main/java/org/zanata/action/HomePage.java
> <https://github.com/zanata/zanata-server/pull/732#discussion_r26544193>:
>
> > +import org.zanata.ApplicationConfiguration;
> > +import org.zanata.events.HomeContentChangedEvent;
> > +
> > +/**
> > + * @author Sean Flanigan <a href="mailto:sflaniga@redhat.com">sflaniga@redhat.com</a>
> > + */
> > +@AutoCreate
> > +@Name("homePage")
> > +@Scope(ScopeType.APPLICATION)
> > +@Slf4j
> > +public class HomePage {
> > +
> > +    @In
> > +    private ApplicationConfiguration applicationConfiguration;
> > +
> > +    @In("commonMarkRenderer")
>
> Any reason for using explicit names?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/732/files#r26544193>.
>

carlosmunoz commented on zanata-war/src/main/java/org/zanata/action/HomePage.java:
```diff
@@ -0,0 +1,67 @@
+
+    private String html;
+
+    public String getHtml() {
```
Can we call this something more descriptive? Perhaps `mainContent` or `homeContent` (since that's exactly what it's fetching)

seanf replied:
Is `homePage.html` not enough?  That's how it will be used.

On 17 March 2015 at 11:54, Carlos Munoz <notifications@github.com> wrote:

> In zanata-war/src/main/java/org/zanata/action/HomePage.java
> <https://github.com/zanata/zanata-server/pull/732#discussion_r26544243>:
>
> > + */
> > +@AutoCreate
> > +@Name("homePage")
> > +@Scope(ScopeType.APPLICATION)
> > +@Slf4j
> > +public class HomePage {
> > +
> > +    @In
> > +    private ApplicationConfiguration applicationConfiguration;
> > +
> > +    @In("commonMarkRenderer")
> > +    private CommonMarkRenderer renderer;
> > +
> > +    private String html;
> > +
> > +    public String getHtml() {
>
> Can we call this something more descriptive? Perhaps mainContent or
> homeContent (since that's exactly what it's fetching)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/732/files#r26544243>.
>

carlosmunoz commented:
Unit or funtional tests for this?
Also, documentation? (although it might need an overal project settings configuration page)

seanf commented:
I have now updated the previous functional tests for these pages to work
with the modifications I made, and modified the user guide's documentation
for the About Project page.

On 17 March 2015 at 12:05, Carlos Munoz <notifications@github.com> wrote:

> Unit or funtional tests for this?
> Also, documentation? (although it might need an overal project settings
> configuration page)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zanata/zanata-server/pull/732#issuecomment-82034833>.
>

carlosmunoz commented:
Changes reviewed :+1:

seanf commented:
@carlosmunoz Do you want to look at the Liquibase code?

carlosmunoz commented:
:+1:

definite commented:
@zanata-jenkins retest this please

definite commented:
@zanata-jenkins retest this please

definite commented:
Tested with text:
    My code doesn't work:

    `for (int i=0; i<100; i--)
    {
        doSomething();
    }`

   Any ideas?

And it converted to 

    <!-- The following text was converted from Seam Text to CommonMark by Zanata.  Some formatting changes may have occurred. -->

    My code doesn't work:

    ```
    for (int i=0; i&lt;100; i--)
    {
         doSomething();
    }
    ```


    Any ideas?

While the preview (Setting->about) render correctly, but in the actual About page, the code squeeze into one line.

seanf commented:
@definite thanks.  Fortunately, I think it's just a problem with the rendering, not with the conversion.  That should make it easier for me to fix and for you to test the fix.

definite commented:
Also table rendered good in preview, but it squeezed as one line in About tab

    <table>
        <tr><td>First name:</td><td>Gavin</td></tr>
        <tr><td>Last name:</td><td>King</td></tr>
    </table>

seanf commented:
@definite I have whitelisted some more HTML tags, so please grab the latest build.

seanf commented:
Squash-merged as c970d50b3bd884203446bd44214ee3ffdfb4b33f.

seanf commented:
Release notes for this are in https://github.com/zanata/zanata-server/commit/ee7b454f4ed829cfbd48e90eda142e49434f4d9b .
